Controllers/ToDoController.cs

```diff
@@ -0,0 +1,286 @@
+﻿using RazorPagesAssignment.Data;
+using RazorPagesAssignment.Models;
+using ExcelDataReader;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using X.PagedList;
+
+namespace RazorPagesAssignment.Controllers
+{
+    [Authorize]
+    public class ToDoController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+
+        public ToDoController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public IActionResult Index(string sortOrder, string searchString, int? page)
+        {
+            ViewData["CurrentSort"] = sortOrder;
+            ViewData["TitleSort"] = String.IsNullOrEmpty(sortOrder) ? "Title" : "title_desc";
```

Comment:
create a separate method to initialize the sort order

Reply:
Done